experiments/standalone/Soil/richards_runoff.jl

```diff
@@ -0,0 +1,246 @@
+
+soil_params = ClimaLand.Soil.RichardsParameters(;
+    hydrology_cm = hydrology_cm,
+    ν = ν,
+    K_sat = K_sat,
+    S_s = S_s,
+    θ_r = θ_r,
+)
+
+function precip_function(t)
+    return -1e-6
+end
+precip = ClimaLand.TimeVaryingInput(precip_function)
+atmos = ClimaLand.PrescribedPrecipitation{FT, typeof(precip)}(precip)
+bottom_bc = ClimaLand.Soil.WaterFluxBC((p, t) -> 0.0)
+bc = (;
+    top = ClimaLand.Soil.RichardsAtmosDrivenFluxBC(atmos, runoff_model),
+    bottom = bottom_bc,
+)
+model = ClimaLand.Soil.RichardsModel{FT}(;
+    parameters = soil_params,
+    domain = domain,
+    boundary_conditions = bc,
+    sources = (),
+    lateral_flow = false,
+)
+Y, p, t = initialize(model)
+z = ClimaCore.Fields.coordinate_field(domain.space.subsurface).z
+lat = ClimaCore.Fields.coordinate_field(domain.space.subsurface).lat
+function hydrostatic_profile(
+    lat::FT,
+    z::FT,
+    ν::FT,
+    θ_r::FT,
+    α::FT,
+    n::FT,
+    S_s::FT,
+    fmax,
+) where {FT}
+    m = 1 - 1 / n
+    zmin = FT(-50.0)
+    zmax = FT(0.0)
+
+    z_∇ = FT(zmin / 5.0 + (zmax - zmin) / 2.5 * (fmax - 0.35) / 0.7)
+    if z > z_∇
+        S = FT((FT(1) + (α * (z - z_∇))^n)^(-m))
+        ϑ_l = S * (ν - θ_r) + θ_r
+    else
+        ϑ_l = -S_s * (z - z_∇) + ν
+    end
+    return FT(ϑ_l)
+end
+t0 = 0.0
+tf = 360.0 * 10
+dt = 360.0 * 2
+Y.soil.ϑ_l .= hydrostatic_profile.(lat, z, ν, θ_r, vg_α, vg_α .+ 2, S_s, f_max)
+@. Y.soil.ϑ_l = oceans_to_zero(Y.soil.ϑ_l, mask)
+set_initial_cache! = make_set_initial_cache(model)
+exp_tendency! = make_exp_tendency(model);
+imp_tendency! = ClimaLand.make_imp_tendency(model);
+update_jacobian! = ClimaLand.make_update_jacobian(model);
+
+set_initial_cache!(p, Y, t0)
+stepper = CTS.ARS111()
+norm_condition = CTS.MaximumError(FT(1e-8))
+conv_checker = CTS.ConvergenceChecker(; norm_condition = norm_condition)
+ode_algo = CTS.IMEXAlgorithm(
+    stepper,
+    CTS.NewtonsMethod(
+        max_iters = 1,
+        update_j = CTS.UpdateEvery(CTS.NewNewtonIteration),
+        convergence_checker = conv_checker,
+    ),
+)
+
+# set up jacobian info
+jac_kwargs =
+    (; jac_prototype = RichardsTridiagonalW(Y), Wfact = update_jacobian!)
+
+prob = SciMLBase.ODEProblem(
+    CTS.ClimaODEFunction(
+        T_exp! = exp_tendency!,
+        T_imp! = SciMLBase.ODEFunction(imp_tendency!; jac_kwargs...),
+        dss! = ClimaLand.dss!,
+    ),
+    Y,
+    (t0, tf),
+    p,
+)
+saveat = Array(t0:dt:tf)
+sv = (;
+    t = Array{Float64}(undef, length(saveat)),
+    saveval = Array{NamedTuple}(undef, length(saveat)),
+)
+saving_cb = ClimaLand.NonInterpSavingCallback(sv, saveat)
+updateat = deepcopy(saveat)
+updatefunc = ClimaLand.make_update_drivers(atmos, nothing)
+driver_cb = ClimaLand.DriverUpdateCallback(updateat, updatefunc)
+cb = SciMLBase.CallbackSet(driver_cb, saving_cb)
+@time sol = SciMLBase.solve(prob, ode_algo; dt = dt, saveat = dt, callback = cb)
+
+longpts = range(-180.0, 180.0, 101)
```

Comment:
That's ~ 3 degree spatial resolution? (slightly different for lon and lat)

Reply:
yes! partly because create the TOPMODEL parameter map at higher resolution is very slow. but eventually I think we should make a map at higher resolution (1/2 degree?) and have that be our benchmark run, instead of 3 degrees.